src/components/PageContainer.jsx

```diff
@@ -33,6 +40,25 @@ const PageContainer = (props) => {
     }
   };
 
+  const setUserContext = () => {
+    axios.get(USER_URL, {
+      headers: { Authorization: `Bearer ${Auth.retrieveToken()}` },
+    })
+      .then((resp) => {
+        setUser(resp.data);
+      })
+      .catch((err) => {
+        if (err.response) {
+          switch (err.response.status) {
+            case 401: history.push(`/sign-in?source=${location.pathname.substring(1)}`); break;
+            case 422: history.push(`/sign-in?source=${location.pathname.substring(1)}`); break;
+            case 405: history.push(`/sign-in?source=${location.pathname.substring(1)}`); break;
+            default: history.push(`/sign-in?source=${location.pathname.substring(1)}`);
```

Comment:
You can combine cases together:
```suggestion
            case 401:
            case 422:
            case 405:
              history.push(`/sign-in?source=${location.pathname.substring(1)}`);
              break;
            default:
              history.push(`/sign-in?source=${location.pathname.substring(1)}`);
```

Reply:
Actually, we don't need a `switch` case here as the outcome is always exactly the same as the `default` case :)